n_helpers.c

```diff
@@ -435,39 +514,40 @@ const char * NoteBinaryTransmit(uint8_t * data, size_t dataLen, size_t bufLen,
     // Validate the index provided by the caller, against the `length` value
     // returned from the Notecard to ensure the caller and Notecard agree on
     // how much data is residing on the Notecard.
-    if (offset != len) {
+    if ((long)notecardOffset != len) {
         NOTE_C_LOG_ERROR("notecard data length is misaligned with offset");
         return ERRSTR("notecard data length is misaligned with offset", c_mem);
     }
 
-    // When `offset` is zero, the entire buffer is available
-    const size_t remaining = (offset ? (max - len) : max);
-    if (dataLen > remaining) {
+    // When offset is zero, the entire buffer is available
```

Comment:
There are a lot of buffers being thrown around. I'd specify that by "the entire buffer", we mean "the entire binary buffer on the Notecard."